hwy/highway_export.h

```diff
@@ -0,0 +1,70 @@
+// Pseudo-generated file to handle both cmake & bazel build system.
+
+// Initial generation done using cmake code:
+// include(GenerateExportHeader)
+// generate_export_header(hwy EXPORT_MACRO_NAME HWY_DLLEXPORT EXPORT_FILE_NAME
+// hwy/highway_export.h)
+// code reformatted using clang-format --style=Google
+
+#ifndef HWY_DLLEXPORT_H
+#define HWY_DLLEXPORT_H
+
+// Bazel build are always static:
+#if !defined(HWY_SHARED_DEFINE) && !defined(HWY_STATIC_DEFINE)
+#define HWY_STATIC_DEFINE
+#endif
+
+#ifdef HWY_STATIC_DEFINE
+#define HWY_DLLEXPORT
+#define HWY_NO_EXPORT
+#else
+#ifndef HWY_DLLEXPORT
+#if defined(hwy_EXPORTS) || defined(hwy_contrib_EXPORTS) || defined(hwy_test_EXPORTS)
+/* We are building this library */
+#ifdef _WIN32
+#define HWY_DLLEXPORT __declspec(dllexport)
+#else
+#define HWY_DLLEXPORT __attribute__((visibility("default")))
+#endif
+#else
+/* We are using this library */
+#ifdef _WIN32
+#define HWY_DLLEXPORT __declspec(dllimport)
+#else
+#define HWY_DLLEXPORT __attribute__((visibility("default")))
+#endif
+#endif
+#endif
+
+#ifndef HWY_NO_EXPORT
```

Comment:
I think we can remove the NO_EXPORT, DEPRECATED_EXPORT, DEPRECATED_NO_EXPORT, and NO_DEPRECATED blocks? (DEPRECATED is potentially useful, might move it to base.h but it's fine for now)